tools/testsys/src/wait.rs

```diff
@@ -0,0 +1,285 @@
+    let mut results = HashMap::new();
+    for crd in crds {
+        results.extend(wait_for_crd(client, crd).await?.context()?)
+    }
+
+    Ok(results)
+}
+
+/// Wait until the conflicting resources are deleted.
+async fn wait_for_conflicting_resources(client: &TestManager, resources: &[String]) -> Result<()> {
+    // While any of the conflicting resources still exist
+    while let Some(resource) = fetch_resources(client, resources).await?.pop() {
+        let resource_name = resource.object_name();
+
+        if matches!(
+            resource.spec.destruction_policy,
+            DestructionPolicy::OnDeletion | DestructionPolicy::Never
+        ) && resource.created_resource().is_some()
+        {
+            warn!("A conflicting resource '{resource_name}' is completed, but has 'DestructionPolicy:{}' and will need to be manually deleted.", resource.spec.destruction_policy)
+        }
+
+        if let Some(e) = resource.creation_error() {
+            warn!("A conflicting resource '{resource_name}' had a creation error '{}' and will need to be manually deleted.", e)
+        }
+
+        if let Some(e) = resource.destruction_error() {
+            warn!("A conflicting resource '{resource_name}' had a destruction error '{}' and will need to be manually deleted.", e)
+        }
+
+        // If a resource has been completed, determine why it hasn't been cleaned up yet.
+        if resource.created_resource().is_some()
+            && matches!(
+                resource.destruction_task_state(),
+                TaskState::Running | TaskState::Completed
+            )
+        {
+            let dependent_tests = fetch_tests_using_resource(client, resource_name).await?;
+            if let Some(still_running_test) = dependent_tests.iter().find(|test| {
+                matches!(
+                    test.agent_status().task_state,
+                    TaskState::Running | TaskState::Unknown
+                )
+            }) {
+                info!("A conflicting resource '{resource_name}' is waiting for a dependent test '{}' to be completed.", still_running_test.object_name());
+            }
+            if let Some(failed_test) = dependent_tests
+                .iter()
+                .find(|test| matches!(test.agent_status().task_state, TaskState::Error))
+            {
+                if matches!(
+                    resource.spec.destruction_policy,
+                    DestructionPolicy::OnTestSuccess
+                ) {
+                    warn!("A conflicting resource '{resource_name}' with `DestructionPolicy:OnTestSuccess' is waiting for a failed test '{}' to be deleted. The test will need to be deleted manually.", failed_test.object_name());
+                }
+            }
+        }
+
+        info!(
+            "Still waiting for resource '{resource_name}' to be deleted from the cluster. Sleeping 60s"
+        );
+        tokio::time::sleep(Duration::from_secs(60)).await;
```

Comment:
60 seconds between polls seem a bit long to me. Any reasoning behind this?
Can we capture the duration of the wait in a const can reuse it whereever it applies?

Reply:
60 seconds was arbitrary, but I didn't want to go too short since it could be polling for hours, and 60 seconds didn't seem too bad. I'll bring it out as a constant anyways.